### PR TITLE
cleanup core plugin code

### PR DESCRIPTION
**Pull request by Schlaefer**
tl;dr Clean up plugin code in core. Adds a method to plugins to resolve their file-path. Pass plugins in event.

---

Refactors the plugin loading related blob of code from `Bootstrap::initializePlugins()` into a new class `Plugin\PluginRepository`. 
		
Move anything from the new `Plugin\PluginRepository` that concerns a single plugin to `Plugin\AbstractPlugin`.

We end up with a fleshed out plugin initialization in `Plugin\AbstractPlugin`, a new `AbstractPlugin::getPluginPath()` and a unique property `$pluginKey`. The last two are imho nice to have for plugin-devs.

I made a commit to the phpFastCache plugin that demonstrates `getPluginPath()` usage.

The $pluginKey uniquely identifies the plugin, we can derive class, vendor, path from it and it's ready-to-use as prefix for global namespaces (cache, events, session, …).

Deprecate `Bootstrap::getPlugins()` and `Utility::isPluginLoaded()`. From now on plugins are passed in the `plugins_loaded`-event. This is cleaner (no coupling to Bootstrap, no Utility grab-bag) and more in Phile's current event-driven spirit. The only use of those deprecated methods I found were [1] and [2] and in both cases accessing `plugins_loaded` feels natural.
			
[1]: https://github.com/infostreams/pagelist-snippet/blob/master/Classes/Plugin.php#L52
[2]: https://github.com/PhileCMS/phileRender/blob/master/Classes/Plugin.php#L58

### CODE REVIEW

**Comment by Stijn-Flipper** [comment]
:+1: You're doing a great job!

**Comment by Schlaefer** [comment]
Last call for raising concerns before I merge this.
